interp/src/structures/names.rs

```diff
@@ -42,6 +43,23 @@ impl Eq for InstanceName {}
 #[derive(Debug, Clone, Hash, PartialEq, Eq)]
 pub struct ComponentQIN(Rc<Vec<InstanceName>>);
 
+impl ComponentQIN {
+    pub fn as_id(&self) -> Id {
+        let mut acc = String::new();
+        match self.0.len() {
+            0 => {}
+            _n => {
+                write!(acc, "{}", &self.0[0].instance)
+                    .expect("error with name?");
+                for i in self.0.iter().skip(1) {
```

Comment:
You can probably use `.join` defined on Vectors of String instead of using `write!`.

Reply:
Ah, I always forget this exists